src/qunit.cpp

```diff
@@ -284,6 +286,91 @@ template <typename F, typename... B> void QUnit::EntangleAndCallMemberRot(F fn,
     ((*qbits).*fn)(radians, bits...);
 }
 
+template <typename CF, typename F>
+void QUnit::ControlCallMember(CF cfn, F fn, bitLenInt control, bitLenInt target, bool anti)
+{
+    real1 prob = Prob(control);
+    if (anti) {
+        prob = ONE_R1 - prob;
+    }
+    if (prob <= REAL_CLAMP) {
```

Comment:
Does this need to be abs()'ed?

Reply:
Oh, I assume not because probabilities shouldn't be negative :P . Habitual question from my day's writing graphics code :)

Reply:
Yes, the probability won't be negative. There is some float rounding error, but it won't affect us, here.